src/gluonts/dataset/common.py

```diff
@@ -216,10 +221,36 @@ def is_valid(cls, path: Path) -> bool:
         return not (path.name.startswith(".") or path.name == "_SUCCESS")
 
 
+class InMemoryDataset(Dataset):
```

Comment:
I'm a bit torn on this. Ideally, we should just get rid of the `ListDataset` and move the `ProcessDataEntry` code downstream to the models.

Then you can just use `dataset = [...]` as your dataset, and don't need to wrap it into `ListDataset`.

Reply:
I'll just remove it for now, should be part of another PR, I guess.